config/config.go

```diff
@@ -3,48 +3,197 @@ package config
+	User     string `yaml:"user"     env:"USER" envDefault:"postgres"`
+	Password string `yaml:"password" env:"PASS"`
+}
+
+type AllConfig struct {
+	Instance       InstanceConfig `yaml:"instance"             envPrefix:"INSTANCE_"`
+	Email          EmailConfig    `yaml:"email"                envPrefix:"EMAIL_"`
+	Database       DatabaseConfig `yaml:"database"             envPrefix:"DB_"`
+	TorProxy       string         `yaml:"tor_proxy"            env:"TOR_PROXY"`
+	CookieKey      string         `yaml:"cookie_key,omitempty" env:"COOKIE_KEY"`
+	AuthReq        []string       `yaml:"auth_req"             env:"AUTH_REQ"        envSeparator:","`
+	PostPerPage    int            `yaml:"posts_per_page"       env:"POSTS_PER_PAGE"                   envDefault:"10" `
+	SupportedFiles []string       `yaml:"supported_files"      env:"SUPPORTED_FILES" envSeparator:"," envDefault:"image/gif,image/jpeg,image/png,image/webp,image/png,video/mp4,video/ogg,video/webm,audio/mpeg,audio/ogg,audio/wav,audio/wave,audio/x-wav"`
+	ModKey         string         `yaml:"mod_key"              env:"MOD_KEY"`
+}
+
+var ConfigFile string
+var configData AllConfig
+
+var Port string
+var TP string
+var Domain string
+var InstanceName string
+var InstanceSummary string
+var Salt string
+var MaxFilesize int64
+
+var SiteEmail string //contact@fchan.xyz
+var SiteEmailPassword string
+var SiteEmailServer string //mail.fchan.xyz
+var SiteEmailPort string   //587
+var SiteEmailNotifyTo string
+
+var DBHost string
+var DBPort int
+var DBUser string
+var DBPassword string
+var DBName string
+
+var TorProxy string //127.0.0.1:9050
+var CookieKey string
+var AuthReq []string
+var PostCountPerPage int
+var SupportedFiles []string
+var Key string
+
 var ActivityStreams = "application/ld+json; profile=\"https://www.w3.org/ns/activitystreams\""
-var AuthReq = []string{"captcha", "email", "passphrase"}
-var PostCountPerPage = 10
-var SupportedFiles = []string{"image/gif", "image/jpeg", "image/png", "image/webp", "image/apng", "video/mp4", "video/ogg", "video/webm", "audio/mpeg", "audio/ogg", "audio/wav", "audio/wave", "audio/x-wav"}
 var Log = log.New(os.Stdout, "", log.Ltime)
 var MediaHashs = make(map[string]string)
-var Key = GetConfigValue("modkey", "")
 var Themes []string
 var DB *sql.DB
 
-// TODO Change this to some other config format like YAML
-// to save into a struct and only read once
-func GetConfigValue(value string, ifnone string) string {
+func LoadConfig() {
+
+	filePath, set := os.LookupEnv("CONFIG")
+	err := env.Parse(&configData)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	if set {
+		loadFromYAML(filePath)
+	} else {
+		filePath = "config/config-init.yaml" // Default to this path
+
+		if _, err := os.Stat(filePath); os.IsNotExist(err) {
+			if info, err := os.Stat("config/config-init"); !os.IsNotExist(err) {
+				// If config/config-init.yaml doesn't exist, but config/config-init does, run migration
+				migrateOldConfig()
```

Comment:
configData = migrateOldConfig() after previous function is converted. easier to read follow